internal/reconciler/telemetry/reconciler.go

```diff
@@ -210,3 +218,72 @@ func (r *Reconciler) serverSideApply(ctx context.Context, obj client.Object) err
 	obj.SetResourceVersion("")
 	return r.Patch(ctx, obj, client.Apply, client.ForceOwnership, client.FieldOwner(fieldOwner))
 }
+
+func (r *Reconciler) checkAnyResourceExist(ctx context.Context) bool {
+	return r.checkLogParserExist(ctx) ||
+		r.checkLogPipelineExist(ctx) ||
+		r.checkMetricPipelinesExist(ctx) ||
+		r.checkTracePipelinesExist(ctx)
+}
```

Comment:
Let's fix the wording here - either CustomResourcesExist or CustomResourceExists